### PR TITLE
[HOPSWORKS-1372] Upgrade the ELK stack to 7.2 OSS and use the OpenDis…

### PR DESCRIPTION
**Pull request by maismail**
…tro Security Plugin
https://logicalclocks.atlassian.net/browse/HOPSWORKS-1372
tests: https://github.com/logicalclocks/hops-testing/pull/565

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -58,3 +60,61 @@
     --es.indices
     --es.shards
 ]
+
+#OpenDistro Security Plugin
+default['elastic']['opendistro']['version'] = "1.2.0.0"
```
Comment (SirOibaf) [inline attributes/default.rb]: Please tabulate the attributes.

File: libraries/elastic_helpers.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    def all_elastic_urls_str()
+      return all_elastic_urls().map{|e| "\"#{e}\""}.join(",")
+    end
+
+    def get_elastic_url(elastic_ip)
+      my_ip = my_private_ip()
+      if opendistro_security?() && opendistro_security_https?()
+        if my_ip.eql? elastic_ip
+          return "https://#{node['fqdn']}:#{node['elastic']['port']}"
```
Comment (SirOibaf) [inline libraries/elastic_helpers.rb]: Does it work if we deploy in localhost mode? Will elastic be listening on 0.0.0.0?

File: providers/start.rb [hunk 3]
```diff
@@ -1,388 +1,368 @@
...
-  EOF
-  end
-
-end
-
-numRetries=10
-retryDelay=20
-
+  if new_resource.systemd == true
+    bash 'elastic-start-systemd' do
```
Comment (SirOibaf) [inline providers/start.rb]: You don't need these anymore. You should use the provider in Kagent to restart these services.

File: providers/start.rb [hunk 4]
```diff
@@ -1,388 +1,368 @@
...
+      code <<-EOF
+      systemctl daemon-reload
+      systemctl stop elasticsearch
+      systemctl start elasticsearch
+    EOF
+    end
 
-Chef::Log.info  "Elastic Ip is: http://#{new_resource.elastic_ip}:#{node['elastic']['port']}"
+  else
+    bash 'elastic-start-systemv' do
```
Comment (SirOibaf) [inline providers/start.rb]: Nowadays we are heavily dependent on Systemd. I don't see the reason of maintaining any other init system.

File: attributes/default.rb [hunk 5]
```diff
@@ -58,3 +60,61 @@
...
+default['elastic']['opendistro_security']['enabled'] = "true"
+default['elastic']['opendistro_security']['url'] = "#{node['download_url']}/opendistro_security-#{node['elastic']['opendistro']['version']}.zip"
+default['elastic']['opendistro_security']['base_dir'] = "#{node['elastic']['plugins_dir']}/opendistro_security"
+default['elastic']['opendistro_security']['config_dir'] = "#{node['elastic']['opendistro_security']['base_dir']}/securityconfig"
+default['elastic']['opendistro_security']['tools_dir'] = "#{node['elastic']['opendistro_security']['base_dir']}/tools"
+default['elastic']['opendistro_security']['tools']['hash'] = "#{node['elastic']['opendistro_security']['tools_dir']}/hash.sh"
+default['elastic']['opendistro_security']['tools']['securityadmin'] = "#{node['elastic']['opendistro_security']['tools_dir']}/securityadmin.sh"
+
+default['elastic']['opendistro_security']['admin']['username'] = "admin"
```
Comment (SirOibaf) [inline attributes/default.rb]: Any way we can autogenerate these passwords? They are only used internally as far as I understood.

File: attributes/default.rb (hunk 1 above)
Comment (maismail) [reply attributes/default.rb]: Ok

File: libraries/elastic_helpers.rb (hunk 2 above)
Comment (maismail) [reply libraries/elastic_helpers.rb]: Will try it on a vm to confirm.

File: attributes/default.rb (hunk 5 above)
Comment (maismail) [reply attributes/default.rb]: ok, will try the random_password from openssl 
https://supermarket.chef.io/cookbooks/openssl/versions/8.5.5

File: providers/start.rb (hunk 3 above)
Comment (maismail) [reply providers/start.rb]: It is very old piece of code, will rewrite to use the kagent_config.

File: providers/start.rb (hunk 4 above)
Comment (maismail) [reply providers/start.rb]: Not sure, if it is every used will remove it then

File: attributes/default.rb (hunk 5 above)
Comment (SirOibaf) [reply attributes/default.rb]: Actually we need to do it also for other passwords, maybe we can have a separate PR. We need to be careful to handle upgrades correctly

Comment (maismail) [reply attributes/default.rb]: I agree, it turned out not to be trivial to fix

**Pull request merged**